src/terminal.c

```diff
@@ -465,6 +465,9 @@ ex_terminal(exarg_T *eap)
 	cmd = skipwhite(p);
     }
 
+    if (cmd == NULL || *cmd == NUL)
+	cmd = p_sh;
```

Comment:
I think this causes https://github.com/vim/vim/issues/1974: if there's an autocommand (e.g. BufEnter) that changes `&shell` when term_start() opens a new window, `cmd` will be pointing to freed memory.